echo_test.go

```diff
@@ -17,11 +17,114 @@
 package main
 
 import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
 const prefixURI = "/v0"
 
 func TestInvalidMethod(t *testing.T) {
 
+	req, err := http.NewRequest("GET", "/v0/echo", nil)
+
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	rr := httptest.NewRecorder()
+	handler := EchoHandler{}
+	handler.ServeHTTP(rr, req)
+
+	assert.Equal(t, http.StatusMethodNotAllowed, rr.Code, "Incorrect status code returned")
+
+}
+
+func TestBadRequest(t *testing.T) {
```

Comment:
`TestBadRequest`, `TestBadBody`, and `TestBody` all test for conditions that should return an `http.StatusBadRequest`.  Since this one is testing what happens when a zero-length body is send, maybe name it `TestEmptyBody`?